src/gcodePlanner.cpp

```diff
@@ -515,11 +528,11 @@ int GCodePlanner::spiralizeWallSlice(GCodePathConfig* config, PolygonRef wall, P
         }
         // now interpolate between last_p and p depending on how far we have progressed along wall
 #if 0
-        if(i == n_points)
+        if(index == n_points)
             addTravel_simple(p);
         else
 #endif
-        addExtrusionMove(Point(last_p + (p - last_p) * static_cast<double>(i) / n_points), config, SpaceFillType::Polygons, 1.0, true);
+        addExtrusionMove(last_p + (p - last_p) * (wall_length / total_length), config, SpaceFillType::Polygons, 1.0, true);
```

Comment:
Again, you could have measured `wall_length` and `total_length` in integer logic and omitted the brackets on this line to achieve the same result.
The way you did it is also good.